tiledb/tests/test_libtiledb.py

```diff
@@ -2730,21 +2730,42 @@ def test_array_vacuum(self):
         path = self.path("test_array_vacuum")
 
         dshape = (0, 19)
-        dom = tiledb.Domain(tiledb.Dim(ctx=ctx, domain=dshape, tile=3), ctx=ctx)
-        att = tiledb.Attr(ctx=ctx, dtype='int64')
-        schema = tiledb.ArraySchema(ctx=ctx, domain=dom, attrs=(att,))
-        tiledb.libtiledb.Array.create(path, schema)
+        num_writes = 10
 
-        for i in range(*dshape):
-            with tiledb.open(path, 'w') as A:
-                A[i:dshape[1]] = np.random.rand(dshape[1] - i)
+        def create_array(target_path):
+            dom = tiledb.Domain(tiledb.Dim(ctx=ctx, domain=dshape, tile=3), ctx=ctx)
+            att = tiledb.Attr(ctx=ctx, dtype='int64')
+            schema = tiledb.ArraySchema(ctx=ctx, domain=dom, attrs=(att,))
+            tiledb.libtiledb.Array.create(target_path, schema)
 
+        def write_fragments(target_path):
+            for i in range(num_writes):
+                with tiledb.open(target_path, 'w') as A:
+                    A[i:dshape[1]] = np.random.rand(dshape[1] - i)
+
+        create_array(path)
+        write_fragments(path)
         paths = vfs.ls(path)
-        self.assertEqual(len(paths), 2 * dshape[1] + 3)
+        self.assertEqual(len(paths), 3 + 2 * num_writes)
 
         tiledb.consolidate(path, ctx=ctx)
         tiledb.vacuum(path, ctx=ctx)
-        self.assertEqual(len(paths), 2 * dshape[1] + 3)
+
+        paths = vfs.ls(path)
+        self.assertEqual(len(paths), 5)
+
+        del path
+
+        path2 = self.path("test_array_vacuum_fragment_meta")
+        create_array(path2)
+        write_fragments(path2)
+        tiledb.consolidate(path2,
+                           config=tiledb.Config({'sm.consolidation.mode': 'fragment_meta'}))
+        tiledb.vacuum(path2,
+                      config=tiledb.Config({'sm.vacuum.mode': 'fragment_meta'}))
+        paths = vfs.ls(path2)
+
+        self.assertEqual(len(paths), 3 + 2 * num_writes + 1)
```

Comment:
@stavrospapadopoulos double-checking if this is correct that there should only be one additional `.meta` file after `fragment_meta` consolidation?